src/GroupResolverHandlerInterface.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\og;
+
+/**
+ * @file
+ * Contains \Drupal\og\GroupResolverHandlerInterface.
+ */
+
+/**
+ * Interface OgContextHandlerInterface.
+ *
+ * @package Drupal\og
+ */
+interface GroupResolverHandlerInterface {
```

Comment:
OK so this is a service that is responsible for looping over the context plugins and resolving the group that best matches the current state. It's a bit strange that most of the methods here are directly related to retrieving and updating plugins, which sounds like the job of the plugin manager to me.